src/dataflow-types/src/client/controller.rs

```diff
@@ -56,6 +60,71 @@ pub struct OrchestratorConfig {
     pub storage_addr: String,
 }
 
+fn deserialize_memory_limit<'de, D>(deserializer: D) -> Result<Option<ByteSize>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let maybe_string = Option::<String>::deserialize(deserializer)?;
+    match maybe_string {
+        None => Ok(None),
+        Some(string) => match ByteSize::from_str(&string) {
+            Ok(bs) => Ok(Some(bs)),
+            Err(_e) => {
+                use serde::de::Error;
+                Err(D::Error::invalid_value(
+                    serde::de::Unexpected::Str(&string),
+                    &"valid size in bytes",
+                ))
+            }
+        },
+    }
+}
+
+#[derive(Copy, Clone, Debug, Deserialize)]
+pub struct ClusterReplicaSizeConfig {
+    #[serde(deserialize_with = "deserialize_memory_limit")]
+    memory_limit: Option<ByteSize>,
+    cpu_limit: Option<CpuLimit>,
+    processes: NonZeroUsize,
+}
+
+#[derive(Clone, Debug)]
+pub struct ClusterReplicaSizeMap(pub HashMap<String, ClusterReplicaSizeConfig>);
```

Comment:
Can you just derive `Deserialize` here and then attach the [`flatten`](https://serde.rs/attr-flatten.html) attr to the inner field?

Reply:
TIL!

Reply:
Actually, newtypes flatten by default, so I can just delete the deserialize impl and derive it.